include/quic/connection.hpp

```diff
@@ -90,6 +91,12 @@ namespace oxen::quic
             }));
         }
 
+        template <typename StreamT = Stream, std::enable_if_t<std::is_base_of_v<Stream, StreamT>, int> = 0>
+        std::shared_ptr<StreamT> get_stream(int64_t id)
+        {
+            return std::static_pointer_cast<StreamT>(get_stream_impl(id));
```

Comment:
This needs to be a dynamic_pointer_cast if you can't guarantee that the pointer is actually a StreamT